ui/ozone/platform/wayland/test/mock_xdg_surface.cc

```diff
@@ -0,0 +1,188 @@
+
+#include <xdg-shell-unstable-v6-server-protocol.h>
+
+#include "ui/ozone/platform/wayland/test/mock_xdg_popup.h"
+#include "ui/ozone/platform/wayland/test/mock_xdg_surface.h"
+#include "ui/ozone/platform/wayland/test/test_positioner.h"
+
+namespace wl {
+
+// xdg_surface, zxdg_surface_v6 and zxdg_toplevel shared methods.
+
+void SetTitle(wl_client* client, wl_resource* resource, const char* title) {
+  GetUserDataAs<MockXdgSurface>(resource)->SetTitle(title);
+}
+
+void SetAppId(wl_client* client, wl_resource* resource, const char* app_id) {
+  GetUserDataAs<MockXdgSurface>(resource)->SetAppId(app_id);
+}
+
+void Move(wl_client* client,
+          wl_resource* resource,
+          wl_resource* seat,
+          uint32_t serial) {
+  GetUserDataAs<MockXdgSurface>(resource)->Move(serial);
+}
+
+void Resize(wl_client* client,
+            wl_resource* resource,
+            wl_resource* seat,
+            uint32_t serial,
+            uint32_t edges) {
+  GetUserDataAs<MockXdgSurface>(resource)->Resize(serial, edges);
+}
+
+void AckConfigure(wl_client* client, wl_resource* resource, uint32_t serial) {
+  GetUserDataAs<MockXdgSurface>(resource)->AckConfigure(serial);
+}
+
+void SetWindowGeometry(wl_client* client,
+                       wl_resource* resource,
+                       int32_t x,
+                       int32_t y,
+                       int32_t width,
+                       int32_t height) {
+  GetUserDataAs<MockXdgSurface>(resource)->SetWindowGeometry(x, y, width,
+                                                             height);
+}
+
+void SetMaximized(wl_client* client, wl_resource* resource) {
+  GetUserDataAs<MockXdgSurface>(resource)->SetMaximized();
+}
+
+void UnsetMaximized(wl_client* client, wl_resource* resource) {
+  GetUserDataAs<MockXdgSurface>(resource)->UnsetMaximized();
+}
+
+void SetFullscreen(wl_client* client,
+                   wl_resource* resource,
+                   wl_resource* output) {
+  GetUserDataAs<MockXdgSurface>(resource)->SetFullscreen();
+}
+
+void UnsetFullscreen(wl_client* client, wl_resource* resource) {
+  GetUserDataAs<MockXdgSurface>(resource)->UnsetFullscreen();
+}
+
+void SetMinimized(wl_client* client, wl_resource* resource) {
+  GetUserDataAs<MockXdgSurface>(resource)->SetMinimized();
+}
+
+const struct xdg_surface_interface kMockXdgSurfaceImpl = {
+    &DestroyResource,    // destroy
+    nullptr,             // set_parent
+    &SetTitle,           // set_title
+    &SetAppId,           // set_app_id
+    nullptr,             // show_window_menu
+    &Move,               // move
+    &Resize,             // resize
+    &AckConfigure,       // ack_configure
+    &SetWindowGeometry,  // set_window_geometry
+    &SetMaximized,       // set_maximized
+    &UnsetMaximized,     // set_unmaximized
+    &SetFullscreen,      // set_fullscreen
+    &UnsetFullscreen,    // unset_fullscreen
+    &SetMinimized,       // set_minimized
+};
+
+// zxdg_surface specific interface
```

Comment:
and here

Reply:
Just to confirm, do you want to have two blank lines above the comment?

Reply:
I'd like to have zero blank lines

Reply:
After discussing this, agreed that these comments have lost their value and may be removed.